app/Livewire/LocalIndicatorQuestionForm.php

```diff
@@ -92,25 +99,49 @@ public function form(Form $form): Form
             ];
         })->toArray();
 
-        return $form->statePath('data')
-            ->model($this->xlsformModuleVersion)
-            ->schema([
-                Repeater::make('surveyRows')
-                    ->relationship('surveyRows')
-                    ->deleteAction(fn($action) => $action->button()->label('Delete Question'))
-                    ->reorderableWithDragAndDrop()
-                    ->reorderAction(fn($action) => $action->button()->label('Drag to Reorder Questions'))
-                    ->addActionLabel('Add Question')
-                    ->label('')
-                    ->itemLabel(fn(array $state) => $state['name'] ?? 'New Question')
-                    ->orderColumn('row_number')
-                    ->schema([
+
+        return $table
+            ->query(
+                fn() => SurveyRow::query()
+                    ->where('xlsform_module_version_id', $this->xlsformModuleVersion->id)
+                    // sort record by row_number, to reflect the user defined ordering by drag and drop
+                    ->orderBy('row_number'),
+            )
+            ->columns([
+                TextColumn::make('type')->label('Question Type'),
+                TextColumn::make('name')->label('Variable Name'),
+                TextColumn::make('defaultLabel')->label('Default Label'),
+            ])
+            // allow user to change the ordering by drag and drop
+            ->reorderable('row_number')
+
+            // add "create new record" button in table header
+            ->headerActions([
+                CreateAction::make()
+                    ->label('ADD QUESTION')
+                    ->icon('heroicon-m-plus')
+                    ->button()
+                    ->color('danger')
+                    ->modalWidth(MaxWidth::SevenExtraLarge)
+                    ->modalHeading('ADD QUESTION')
+                    ->createAnother(false)
+                    ->mutateFormDataUsing(function (array $data): array {
+                        // find the largest row_number of survey_rows records
+                        $xlsformModuleVersion = XlsformModuleVersion::find($this->xlsformModuleVersion->id);
+                        $lastRowNumber = $xlsformModuleVersion->surveyRows == null ? $xlsformModuleVersion->surveyRows->last()->row_number : 0;
```

Comment:
Is this logic the wrong way round? All my questions are getting a `row_number` of 1 until I manually re-order them.

Reply:
Just fixed this logical error, the condition should be != null instead of == null